hypothesis-python/tests/cover/test_lookup.py

```diff
@@ -116,6 +122,10 @@ class Elem:
     pass
 
 
+# To avoid SmallSearchSpaceWarning
+st.register_type_strategy(Elem, st.builds(Elem))
```

Comment:
I think `SmallSearchSpaceWarning` should only be emitted if `signature(typ)` has parameters other than `self` (without caring about the name).

Reply:
> I think `SmallSearchSpaceWarning` should only be emitted if `signature(typ)` has parameters other than `self` (without caring about the name).

Agreed. I did actually consider this, but skipped it since I didn't know off-hand how to write a simple-yet-robust check for this. I'm sure I can figure it out, but maybe you already know?

Reply:
We have an internal `get_signature()` helper, then check whether the `.parameters` attribute is empty.

Reply:
Thanks, done. A few lines above it is indicated that `get_signature.parameters` might also have an item for `return`, but I couldn't see that behaviour in a quick test. So I'm ignoring that possibility, just checking for empty `.parameters`.

https://github.com/HypothesisWorks/hypothesis/blob/4a2207619aa2c91f7b1ca376551d67d3bdbc6b27/hypothesis-python/src/hypothesis/strategies/_internal/core.py#L1255

Reply:
Thanks for the review @Zac-HD! I think all issues you raised have been addressed, with only the question above potentially remaining.

Reply:
Yeah, we can ignore that - it's only possible if you've manually constructed and assigned an illegal signature, because using `return` as an argument name is of course a syntax error.  We'll be able to drop that check at the 3.10 EOL in 2026, thanks to https://github.com/python/cpython/pull/92065.